proto/algorithms.proto

```diff
@@ -0,0 +1,13 @@
+syntax = "proto3";
+
+package bayesmix;
+
+enum Algorithm {
```

Comment:
I don't like these enum names (including Hierarchy and Mixing). When we write `bayesmix::Algorithm` it seems like it's an algorithm object. They should be something like AlgorithmChoice or similar